modules/openapi-generator/src/main/resources/Java/libraries/restclient/ApiClient.mustache

```diff
@@ -156,9 +156,9 @@ public class ApiClient{{#jsr310}} extends JavaTimeFormatter{{/jsr310}} {
 
         {{/withXml}}
         Consumer<List<HttpMessageConverter<?>>> messageConverters = converters -> {
-            converters.add(new MappingJackson2HttpMessageConverter(mapper));
+            converters.add(0, new MappingJackson2HttpMessageConverter(mapper));
             {{#withXml}}
-            converters.add(new MappingJackson2XmlHttpMessageConverter(xmlMapper));
+            converters.add(0, new MappingJackson2XmlHttpMessageConverter(xmlMapper));
```

Comment:
This is wrong fix. You need to write test. Now, If set flag `withXml=true` - you will replace JSON converter - this is wong logic

Reply:
@altro3 i did not change anything regarding the logic. In Java when you call with an index, it

* inserts element at specified index
* shifts existing elements to the right
*
So, no replacement, just adding on top priority.

I could have saved existing list content, replace existing converters and put back modified list content yet I would have found harder than inserting at the beginning.

What do you think?

Reply:
Yes, I apologize. This is not a replacement, it is an insert with a shift to the right.

Reply:
no pb :pray: